portal/src/graphql/portal/DesignScreen/form.ts

```diff
@@ -679,7 +679,33 @@ export function useBrandDesignForm(appID: string): BranchDesignForm {
     (color: CSSColor | undefined, targetTheme: Theme) => {
       resourceMutator.updateCustomisableTheme((prev) => {
         return produce(prev, (draft) => {
+          if (color == null) {
+            draft.link.color = undefined;
+            draft.link.colorActive = undefined;
+            draft.link.colorHover = undefined;
+            return;
+          }
           draft.link.color = color;
+          const themeRules = themeRulesStandardCreator();
+          const colorObject = getColorFromString(color);
+          if (colorObject == null) {
+            draft.link.color = undefined;
+            draft.link.colorActive = undefined;
+            draft.link.colorHover = undefined;
+            return;
+          }
+          ThemeGenerator.insureSlots(themeRules, false);
+          ThemeGenerator.setSlot(
+            themeRules[BaseSlots[BaseSlots.primaryColor]],
+            colorObject,
+            false,
+            true,
+            true
+          );
+          const json = ThemeGenerator.getThemeAsJson(themeRules);
```

Comment:
This tedious color generation logic is duplicate with the primary button.

Please introduce a new function

```typescript
function deriveColors(color: string): { original: string, variant: string} | null

...

const derivedColors = deriveColors(color);
if (derivedColors == null) {
  // Set undefined
  return;
}
draft.link.color = derivedColors.original;
draft.link.colorActive = derivedColors.variant;
draft.link.colorHover = derivedColors.variant;
```

Reply:
updated to use existing `getShades` function to get the dark color 🙏